src/core/core/service_api_c.cpp

```diff
@@ -287,6 +292,51 @@ inline void dsn_global_init()
     dsn::service_engine::instance();
 }
 
+static void print_header(FILE *fp, dsn_log_level_t log_level)
```

Comment:
```suggestion
static std::string dsn_log_prefixed_message() {
  ...
}
```

We should not assume the underlying logger using `fprintf(FILE*)` for printing.

Reply:
return std::string is not a good idea. If the string you get from `dsn_log_prefixed_message` is `%sab%dc`, it would occur a disater when you print it using `fprintf(fp, "%s", string);`

Reply:
fixed